### ISSUE TITLE
When an auto-obsoleted update is edited with a new build, it does not go back to testing and no koji-build-group.build.complete message is published

### ISSUE THREAD

**Issue by AdamWill** [fedora-infra/bodhi#5604]
Just noticed this when fixing https://bodhi.fedoraproject.org/updates/FEDORA-2024-c0e4857caa . It got auto-unpushed for -3 karma. I edited the update with a new build, and its karma was reset, but it stayed in obsolete state, and no koji-build-group.build.complete message was published (which means tests were not retriggered). I could manually request it for testing with the CLI (it was not possible in the web UI), but that still didn't publish the message. I had to trigger the openQA tests manually.

I'll look into this myself when I can, but wanted to file it so it's not forgotten...

**Comment by mattiaverga** [comment]
Maybe it might be better to unpush an update that reaches the negative karma, rather than obsolete it? But then we have to be sure it cannot be pushed again if builds aren't updated.

**Comment by AdamWill** [comment]
yeah, I'm not really sure what would be best, was gonna consider all the options.

**Comment by mattiaverga** [comment]
Another case is https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2024-c154b725ab
